doc/fundamentals/creating_analysis_pipelines.md

```diff
@@ -0,0 +1,62 @@
+# Creating Analysis Pipelines
+There are three steps to running an analysis in a Pachyderm "pipeline":
+
+1. Write your code.
+2. Build a [Docker](https://docs.docker.com/engine/getstarted/step_four/) image that includes your code and dependencies.
+3. Create a Pachyderm "pipeline" referencing that Docker image.
+
+Multi-stage pipelines (e.g., parsing -> modeling -> output) can be created by repeating these three steps to build up a graph of processing steps.  For more tips on composing pipelines see ["Composing More Complicated Pipelines"](../cookbook/composing_pipelines_PLACEHOLDER).
+
+## 1. Writing your analysis code
+
+Code used to process data in Pachyderm can be written using any languages or libraries you want. It can be as simple as a bash command or as complicated as a TensorFlow neural network.  At the end of the day, all your code and dependencies will be built into a container that can run anywhere (including inside of Pachyderm). We've got demonstrative [examples on GitHub](https://github.com/pachyderm/pachyderm/tree/master/doc/examples) using bash, Python, TensorFlow, and OpenCV and we're constantly adding more.
+u
+As we touch on briefly in the [beginner tutorial](../getting_started/beginner_tutorial), your code itself only needs to read and write files from a local file system. It does NOT have to import any special Pachyderm functionality or libraries.  You just need to be able to read files and write files.
+
+For the reading files part, Pachyderm automatically mounts each input data repository as `/pfs/<repo_name>` in the running instances of your Docker image (called "containers"). The code that you write just needs to read input data from this directory, just like in any other file system.  Your analysis code also does NOT have to deal with data sharding or parallelization as Pachyderm will automatically shard the input data across parallel containers. For example, if you've got four containers running your Python code, Pachyderm will automatically supply 1/4 of the input data to `/pfs/<repo_name>` in each running container. That being said, you also have a lot of control over how that input data is split across containers. Check out our guide on :doc: `parallelization` to see the details of that.
+
+For the writing files part (saving results, etc.), you code simply needs to write to `/pfs/out`. This is a special directory mounted by Pachyderm in all of your running containers. Similar to reading data, your code doesn't have to manage parallelization or sharding, just write data to `/pfs/out` and Pachyderm will make sure it all ends up in the correct place. 
+
+## 2. Building a Docker Image
```

Comment:
We should mention version tagging your images (e.g. dont use :latest) somewhere here. Should also go under best-practices once we create that doc.